app/models/container_deployment.rb

```diff
@@ -6,7 +6,368 @@ class ContainerDeployment < ApplicationRecord
+        "hostedDomain" => authentication.google_hosted_domain,
+        "challenge"    => "false"
+      }
+    when "AuthenticationHtpasswd"
+      options = {
+        "filename" => "/etc/origin/master/htpasswd"
+      }
+    when "AuthenticationLdap"
+      options = {
+        "attributes"   => {"id"                => authentication.ldap_id,
+                           "email"             => authentication.ldap_email,
+                           "name"              => authentication.ldap_name,
+                           "preferredUsername" => authentication.ldap_preferred_user_name},
+        "bindDN"       => authentication.ldap_bind_dn,
+        "bindPassword" => authentication.password,
+        "ca"           => authentication.certificate_authority,
+        "insecure"     => authentication.ldap_insecure.to_s,
+        "url"          => authentication.ldap_url
+      }
+    when "AuthenticationOpenId"
+      options = {
+        "clientID"                       => authentication.userid,
+        "clientSecret"                   => authentication.password,
+        "claims"                         => {"id" => authentication.open_id_sub_claim},
+        "urls"                           => {"authorize" => authentication.open_id_authorization_endpoint, "toekn" => authentication.open_id_token_endpoint},
+        "challenge"                      => "false",
+        "openIdExtraAuthorizeParameters" => authentication.open_id_extra_authorize_parameters,
+        "openIdExtraScopes"              => authentication.open_id_extra_scopes
+      }
+    when "AuthenticationRequestHeader"
+      options = {
+        "challengeURL"                          => authentication.request_header_challenge_url,
+        "loginURL"                              => authentication.request_header_login_url,
+        "clientCA"                              => authentication.certificate_authority,
+        "headers"                               => authentication.request_header_headers,
+        "requestHeaderPreferredUsernameHeaders" => authentication.request_header_preferred_username_headers,
+        "requestHeaderNameHeaders"              => authentication.request_header_name_headers,
+        "requestHeaderEmailHeaders"             => authentication.request_header_email_headers
+      }
+    end
+    {'name' => "example_name", 'login' => "true", 'challenge' => "true", 'kind' => authentication.authtype}.merge! options
+  end
+
+  def extract_public_ip_or_hostname(deployment_node)
+    return deployment_node.address if deployment_node.vm_id.nil?
+    vm = deployment_node.vm
+    hostname = vm.hostnames
+    if hostname.empty?
+      ips = vm.hardware.ipaddresses
+      ips.each do |ip|
+        return ip if public_ip?(ip)
+      end
+    end
+    hostname.first || ips.last
+  end
+
+  def public_ip?(ip)
+    require "socket"
+    return ip unless Addrinfo.tcp(ip, 80).ipv4_private?
+  end
+
+  def generate_automation_params(params)
```

Comment:
It looks like a number of these methods should be private. For instance, `generate_automation_params` looks like it requires `create_deployment` to massage the `params` before calling `generate_automation_params`.

Can you go through this class and see what methods make sense to privatize and move those to private?